### ISSUE TITLE
[Bug]: v2.3.x RSS feeds returning 404 on cover/episodes

### ISSUE THREAD

**Issue by jansendotsh** [advplyr/audiobookshelf#2008]
### Describe the issue

After upgrading to v2.3.x (anything from 2.3.0 to current), I've had a 404 on a number of RSS "episodes" and the covers don't populate. When I downgrade to v2.2.23, the issue doesn't occur and we're all clear. To share an example of an RSS feed:

```
<?xml version="1.0" encoding="UTF-8"?><rss xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:content="http://purl.org/rss/1.0/modules/content/" xmlns:atom="http://www.w3.org/2005/Atom" version="2.0" xmlns:itunes="http://www.itunes.com/dtds/podcast-1.0.dtd" xmlns:psc="http://podlove.org/simple-chapters" xmlns:podcast="https://podcastindex.org/namespace/1.0" xmlns:googleplay="http://www.google.com/schemas/play-podcasts/1.0"><channel><title><![CDATA[(2010) Confession of a Buddhist Atheist]]></title><description><![CDATA[∩╗┐
Does Buddhism require faith? Can an atheist or agnostic follow the Buddha's teachings without believing in reincarnation or organized religion? This is one man's confession. In his classic Buddhism Without Beliefs, Stephen Batchelor offered a profound, secular approach to the teachings of the Buddha that struck an emotional chord with Western audiences. Now, with the same brilliance and boldness of thought, he paints a groundbreaking portrait of the historical BuddhaΓÇötold from the author's unique perspective as a former Buddhist monk and modern seeker. Drawing from the original Pali Canon, the seminal collection of Buddhist discourses compiled after the Buddha's death by his followers, Batchelor shows us the Buddha as a flesh-and-blood man who looked at life in a radically new way. Batchelor also reveals the everyday challenges and doubts of his own devotional journeyΓÇöfrom meeting the Dalai Lama in India, to training as a Zen monk in Korea, to finding his path as a lay teacher of Buddhism living in France. Both controversial and deeply personal, Stephen Batchelor's refreshingly doctrine-free, life-informed account is essential listening for anyone interested in Buddhism.
]]></description><link>http://192.168.164.3:13378/item/ed7a5379-0e4b-4b73-98bd-b2228f7b228e</link><image><url>http://192.168.164.3:13378/feed/ed7a5379-0e4b-4b73-98bd-b2228f7b228e/cover</url><title>(2010) Confession of a Buddhist Atheist</title><link>http://192.168.164.3:13378/item/ed7a5379-0e4b-4b73-98bd-b2228f7b228e</link></image><generator>Audiobookshelf</generator><lastBuildDate>Tue, 15 Aug 2023 04:07:35 GMT</lastBuildDate><atom:link href="http://192.168.164.3:13378/feed/ed7a5379-0e4b-4b73-98bd-b2228f7b228e" rel="self" type="application/rss+xml"/><language>en</language><author>Stephen Batchelor</author><itunes:author>Stephen Batchelor</itunes:author><itunes:summary>∩╗┐
Does Buddhism require faith? Can an atheist or agnostic follow the Buddha&apos;s teachings without believing in reincarnation or organized religion? This is one man&apos;s confession. In his classic Buddhism Without Beliefs, Stephen Batchelor offered a profound, secular approach to the teachings of the Buddha that struck an emotional chord with Western audiences. Now, with the same brilliance and boldness of thought, he paints a groundbreaking portrait of the historical BuddhaΓÇötold from the author&apos;s unique perspective as a former Buddhist monk and modern seeker. Drawing from the original Pali Canon, the seminal collection of Buddhist discourses compiled after the Buddha&apos;s death by his followers, Batchelor shows us the Buddha as a flesh-and-blood man who looked at life in a radically new way. Batchelor also reveals the everyday challenges and doubts of his own devotional journeyΓÇöfrom meeting the Dalai Lama in India, to training as a Zen monk in Korea, to finding his path as a lay teacher of Buddhism living in France. Both controversial and deeply personal, Stephen Batchelor&apos;s refreshingly doctrine-free, life-informed account is essential listening for anyone interested in Buddhism.
</itunes:summary><itunes:type></itunes:type><itunes:image href="http://192.168.164.3:13378/feed/ed7a5379-0e4b-4b73-98bd-b2228f7b228e/cover"/><itunes:owner><itunes:name>Stephen Batchelor</itunes:name><itunes:email></itunes:email></itunes:owner><itunes:explicit>false</itunes:explicit><itunes:block>Yes</itunes:block><googleplay:block>yes</googleplay:block><item><title><![CDATA[(2010) Confession of a Buddhist Atheist]]></title><description><![CDATA[∩╗┐
Does Buddhism require faith? Can an atheist or agnostic follow the Buddha's teachings without believing in reincarnation or organized religion? This is one man's confession. In his classic Buddhism Without Beliefs, Stephen Batchelor offered a profound, secular approach to the teachings of the Buddha that struck an emotional chord with Western audiences. Now, with the same brilliance and boldness of thought, he paints a groundbreaking portrait of the historical BuddhaΓÇötold from the author's unique perspective as a former Buddhist monk and modern seeker. Drawing from the original Pali Canon, the seminal collection of Buddhist discourses compiled after the Buddha's death by his followers, Batchelor shows us the Buddha as a flesh-and-blood man who looked at life in a radically new way. Batchelor also reveals the everyday challenges and doubts of his own devotional journeyΓÇöfrom meeting the Dalai Lama in India, to training as a Zen monk in Korea, to finding his path as a lay teacher of Buddhism living in France. Both controversial and deeply personal, Stephen Batchelor's refreshingly doctrine-free, life-informed account is essential listening for anyone interested in Buddhism.
]]></description><link>http://192.168.164.3:13378/item/ed7a5379-0e4b-4b73-98bd-b2228f7b228e</link><guid isPermaLink="false">http://192.168.164.3:13378/feed/ed7a5379-0e4b-4b73-98bd-b2228f7b228e/item/45e0b921-451a-4ea1-807f-ad8374576d35/Confession of a Buddhist Atheist (2010).m4b</guid><dc:creator><![CDATA[Stephen Batchelor]]></dc:creator><pubDate>Tue, 15 Aug 2023 03:43:22 GMT</pubDate><enclosure url="http://192.168.164.3:13378/feed/ed7a5379-0e4b-4b73-98bd-b2228f7b228e/item/45e0b921-451a-4ea1-807f-ad8374576d35/Confession of a Buddhist Atheist (2010).m4b" length="335122215" type="audio/mp4"/><itunes:author>Stephen Batchelor</itunes:author><itunes:duration>11:46:27.0</itunes:duration><itunes:summary>∩╗┐
Does Buddhism require faith? Can an atheist or agnostic follow the Buddha&apos;s teachings without believing in reincarnation or organized religion? This is one man&apos;s confession. In his classic Buddhism Without Beliefs, Stephen Batchelor offered a profound, secular approach to the teachings of the Buddha that struck an emotional chord with Western audiences. Now, with the same brilliance and boldness of thought, he paints a groundbreaking portrait of the historical BuddhaΓÇötold from the author&apos;s unique perspective as a former Buddhist monk and modern seeker. Drawing from the original Pali Canon, the seminal collection of Buddhist discourses compiled after the Buddha&apos;s death by his followers, Batchelor shows us the Buddha as a flesh-and-blood man who looked at life in a radically new way. Batchelor also reveals the everyday challenges and doubts of his own devotional journeyΓÇöfrom meeting the Dalai Lama in India, to training as a Zen monk in Korea, to finding his path as a lay teacher of Buddhism living in France. Both controversial and deeply personal, Stephen Batchelor&apos;s refreshingly doctrine-free, life-informed account is essential listening for anyone interested in Buddhism.
</itunes:summary><itunes:explicit>false</itunes:explicit><itunes:episodeType/><itunes:season/><itunes:episode/></item></channel></rss>
```

To drill down on this example, the cover and m4b both return a 404...

```
➜ ~ curl -Il http://192.168.164.3:13378/feed/ed7a5379-0e4b-4b73-98bd-b2228f7b228e/item/013e57f8-df20-4e5a-8f66-132e57b0fbc6/Confession%20of%20a%20Buddhist%20Atheist%20\(2010\).m4b
HTTP/1.1 404 Not Found
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, POST, PATCH, PUT, DELETE, OPTIONS
Access-Control-Allow-Headers: *
Access-Control-Allow-Credentials: true
Content-Type: text/plain; charset=utf-8
Content-Length: 9
ETag: W/"9-0gXL1ngzMqISxa6S1zx3F4wtLyg"
Date: Tue, 15 Aug 2023 04:06:41 GMT
Connection: keep-alive
Keep-Alive: timeout=5

➜ ~ curl -Il http://192.168.164.3:13378/feed/ed7a5379-0e4b-4b73-98bd-b2228f7b228e/cover
HTTP/1.1 404 Not Found
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, POST, PATCH, PUT, DELETE, OPTIONS
Access-Control-Allow-Headers: *
Access-Control-Allow-Credentials: true
Content-Type: text/plain; charset=utf-8
Content-Length: 9
ETag: W/"9-0gXL1ngzMqISxa6S1zx3F4wtLyg"
Date: Tue, 15 Aug 2023 04:07:03 GMT
Connection: keep-alive
Keep-Alive: timeout=5
```

Based on the docker logs, there does seem to be some issues with finding the file although it's there when played via ABS web app or mobile app (or even downloading as logged):

```
[2023-08-15 04:05:21] INFO: [RssFeedManager] Opened RSS feed "http://192.168.164.3:13378/feed/ed7a5379-0e4b-4b73-98bd-b2228f7b228e"
[2023-08-15 04:05:28] INFO: [Server] Requesting rss feed ed7a5379-0e4b-4b73-98bd-b2228f7b228e
getEpisodePath= 013e57f8-df20-4e5a-8f66-132e57b0fbc6 undefined
[2023-08-15 04:05:48] ERROR: [RssFeedManager] Feed episode not found 013e57f8-df20-4e5a-8f66-132e57b0fbc6 (RssFeedManager.js:165)
[2023-08-15 04:06:23] INFO: [Server] Requesting rss feed ed7a5379-0e4b-4b73-98bd-b2228f7b228e
[2023-08-15 04:06:28] INFO: [Server] Requesting rss feed ed7a5379-0e4b-4b73-98bd-b2228f7b228e
getEpisodePath= 013e57f8-df20-4e5a-8f66-132e57b0fbc6 undefined
[2023-08-15 04:06:34] ERROR: [RssFeedManager] Feed episode not found 013e57f8-df20-4e5a-8f66-132e57b0fbc6 (RssFeedManager.js:165)
getEpisodePath= 013e57f8-df20-4e5a-8f66-132e57b0fbc6 undefined
[2023-08-15 04:06:41] ERROR: [RssFeedManager] Feed episode not found 013e57f8-df20-4e5a-8f66-132e57b0fbc6 (RssFeedManager.js:165)
[2023-08-15 04:07:35] INFO: [Server] Requesting rss feed ed7a5379-0e4b-4b73-98bd-b2228f7b228e
[2023-08-15 04:10:39] INFO: [LibraryItemController] User "garrett" requested file download at "/audiobooks/Stephen Batchelor/(2010) Confession of a Buddhist Atheist/Confession of a Buddhist Atheist (2010).m4b"
```

Of course, this is with the reverse proxy out of the loop and v2.2.23 works as expected. It seems there's a couple folks on the Discord server with this issue but I wanted to formalize it and create a bug report here. If there's any additional information that I can provide, feel free to let me know and I'll try to get that back to you as soon as possible. 

### Steps to reproduce the issue

1. Setting up a fresh server with latest version (v2.3.x).
2. Configure Audiobook/Podcast library.
3. Scan library.
4. Enable RSS on audiobook/podcast.
5. Try to use RSS feed.

### Audiobookshelf version

v2.3.3

### How are you running audiobookshelf?

Docker

**Comment by slyfoxgh** [comment]
+1 Same problem. Not a single RSS feed works.

**Comment by nichwall** [comment]
Is this the same issue? If so, it's been fixed for the next release
#1948

**Comment by slyfoxgh** [comment]
> Is this the same issue? If so, it's been fixed for the next release #1948

The problem that I experienced is that the actual audio files were not accessible via RSS.

**Comment by slyfoxgh** [comment]
Error log:

```
2023-08-15 03:08:11
INFO
[Server] Requesting rss feed e1a5e285-f5c8-4a55-9f66-675d53680018
2023-08-15 03:08:14
ERROR
[RssFeedManager] Feed episode not found acdb8cd3-2e2d-4158-bd60-695e928e0073
2023-08-15 03:08:14
ERROR
**[RssFeedManager] Feed episode not found** acdb8cd3-2e2d-4158-bd60-695e928e0073
2023-08-15 03:08:17
ERROR
[RssFeedManager] Feed episode not found acdb8cd3-2e2d-4158-bd60-695e928e0073
2023-08-15 03:08:19
ERROR
[RssFeedManager] Feed episode not found acdb8cd3-2e2d-4158-bd60-695e928e0073
2023-08-15 03:08:24
INFO
[RssFeedManager] Closed RSS feed "http://192.168.68.50:13378/feed/e1a5e285-f5c8-4a55-9f66-675d53680018"
2023-08-15 03:08:29
INFO
[PlaybackSessionManager] startSession: Closing open session "Сборник американской фантастики Паломничество на Землю" for user "ivan" (Device: Mac OS 10.15.7 / Safari)
```

ABS v2.3.3

**Comment by theprinceofspace** [comment]
I am having this same issue as well.

**Comment by dev-nicolaos** [comment]
I'm also having this issue. I'm running Audiobookshelf in a PikaPod so I know its not that I screwed up something in the deployment. Was working fine before the upgrade to 2.3, but now no audio tracks can be found when I open an RSS feed. The feed itself has an entry for each track, but the track isn't loaded. Seen with both an M4B audiobook and one with a collection of MP3s

**Comment by advplyr** [comment]
This is fixed in edge already and will be fixed in the upcoming release.

**Comment by advplyr** [comment]
Fixed in v2.4.1
